nixos/tests/phpfpm.nix

```diff
@@ -0,0 +1,71 @@
+import ./make-test-python.nix ({ pkgs, lib, ... }: {
+  name = "phpfpm";
+  meta.maintainers = [ lib.maintainers.aanderse ];
+
+  machine =
+    { config, ... }:
+    { services.httpd.enable = true;
+      services.httpd.adminAddr = "admin@example.com";
+      services.httpd.extraModules = [ "proxy_fcgi" ];
+      services.httpd.extraConfig = ''
+        Alias /site1 /var/www/site1
+        <Directory /var/www/site1>
+          <FilesMatch "\.php$">
+            <If "-f %{REQUEST_FILENAME}">
+              SetHandler "proxy:unix:${config.services.phpfpm.pools.site1.socket}|fcgi://localhost/"
+            </If>
+          </FilesMatch>
+          DirectoryIndex index.php
+          Require all granted
+        </Directory>
+
+        Alias /site2 /var/www/site2
+        <Directory /var/www/site2>
+          <FilesMatch "\.php$">
+            <If "-f %{REQUEST_FILENAME}">
+              SetHandler "proxy:fcgi://127.0.0.1:9000"
+            </If>
+          </FilesMatch>
+          DirectoryIndex index.php
+          Require all granted
+        </Directory>
+      '';
+      services.phpfpm.pools = {
+        site1 = {
+          user = config.services.httpd.user;
+          group = config.services.httpd.group;
+          settings = {
+            "listen.owner" = config.services.httpd.user;
+            "listen.group" = config.services.httpd.group;
+            "pm" = "static";
+            "pm.max_children" = 2;
+          };
+        };
+        site2 = {
+          user = config.services.httpd.user;
+          group = config.services.httpd.group;
+          settings = {
+            "listen" = 9000;
+            "pm" = "static";
+            "pm.max_children" = 2;
+          };
+        };
+      };
+
+      systemd.tmpfiles.rules = [
```

Comment:
can you use the right user and groups here?

Reply:
You'll have to explain why root isn't the right user in this case :-\ I'm not in the habit of making source code mutable or owned by the executing user if there is no need to.

Reply:
Looking at wordpress and friends, I seems quite common to have php code owned by the process executing it. If we look at extending the test to run the php interpreter as another user than the webserver, having the htdoc files owned by the specific php interpreter might make sense to mimic that behaviour.

Reply:
Whenever a user can execute **and** modify its own code in NixOS it is either: a compromise on security for code ease/clarity, the user requires the ability to modify its own code (usually because of configuration, cache, etc...), or the author has maybe just hasn't thought about it. Generally we don't have the problem too much in nixpkgs because most code is immutable, and when we do have mutable code it usually falls into the category of configuration/cache that is required to be mutable.

Perhaps having those two `.php` files packaged in the nix store would make the situation more acceptable and familiar... :thinking: